### PR TITLE
Add unconditioned experimentation for Fast Fetch, for Doubleclick

### PR DESCRIPTION
**Pull request by bradfrizzell**
Modify Fast Fetch eligibility selection to allow for unconditioned experimentation in Doubleclick.

### CODE REVIEW

File: extensions/amp-ad-network-fake-impl/0.1/fake-a4a-config.js [hunk 1]
```diff
@@ -24,9 +24,13 @@
...
+ * @param {!boolean} useRemoteHtml
  * @returns {boolean}
  */
-export function fakeIsA4AEnabled(win, element) {
+export function fakeIsA4AEnabled(win, element, useRemoteHtml) {
+  if (useRemoteHtml) {
+    return false;
+  }
   const a4aRequested = element.getAttribute('data-use-a4a');
   return !!a4aRequested;
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-fake-impl/0.1/fake-a4a-config.js]: Nit: return !useRemoteHml && !!element.getAttribute('data-use-a4a');

File: extensions/amp-ad-network-gmossp-impl/0.1/gmossp-a4a-config.js [hunk 2]
```diff
@@ -25,9 +25,13 @@ const GMOSSP_SRC_A4A_PREFIX_ = 'https://amp.sp.gmossp-sp.jp/_a4a/';
...
+ * @param {!boolean} useRemoteHtml
  * @returns {boolean}
  */
-export function gmosspIsA4AEnabled(win, element) {
+export function gmosspIsA4AEnabled(win, element, useRemoteHtml) {
+  if (useRemoteHtml) {
+    return false;
+  }
   const src = element.getAttribute('src');
   return !!element.getAttribute('data-use-a4a') && !!src &&
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-gmossp-impl/0.1/gmossp-a4a-config.js]: Nit:  return !useREmoteHtml && !element...

File: extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js [hunk 3]
```diff
@@ -19,9 +19,13 @@ const SRC_PREFIX_ = 'https://ib.3lift.com/';
...
+ * @param {!boolean} useRemoteHtml
  * @returns {boolean}
  */
-export function tripleliftIsA4AEnabled(win, element) {
+export function tripleliftIsA4AEnabled(win, element, useRemoteHtml) {
+  if (useRemoteHtml) {
+    return false;
+  }
   let src;
   return !!element.getAttribute('data-use-a4a') && !!(src =
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js]: Ditto

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 4]
```diff
@@ -136,6 +151,12 @@ export class DoubleclickA4aEligibility {
           (getMode(win).localDev || getMode(win).test)) {
         experimentId = MANUAL_EXPERIMENT_ID;
       } else {
+        if ([DOUBLECLICK_UNCONDITIONED_EXPERIMENTS.FF_CANONICAL_CTL,
+          DOUBLECLICK_UNCONDITIONED_EXPERIMENTS.FF_CANONICAL_EXP,
+        ].includes(experimentId)) {
+          return experimentId ==
```
[the remark this answers is not in the document]
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 5]
```diff
@@ -62,6 +62,11 @@ export const DOUBLECLICK_EXPERIMENT_FEATURE = {
   IDENTITY_EXPERIMENT: '21060938',
 };
 
+export const DOUBLECLICK_UNCONDITIONED_EXPERIMENTS = {
```
[the remark this answers is not in the document]
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: done

File: extensions/amp-ad-network-fake-impl/0.1/fake-a4a-config.js (hunk 1 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-fake-impl/0.1/fake-a4a-config.js]: done

File: extensions/amp-ad-network-gmossp-impl/0.1/gmossp-a4a-config.js (hunk 2 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-gmossp-impl/0.1/gmossp-a4a-config.js]: done

File: extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js (hunk 3 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 6]
```diff
@@ -201,10 +229,11 @@ const singleton = new DoubleclickA4aEligibility();
 /**
  * @param {!Window} win
  * @param {!Element} element
+ * @param {!boolean} useRemoteHtml
  * @returns {boolean}
  */
-export function doubleclickIsA4AEnabled(win, element) {
-  return singleton.isA4aEnabled(win, element);
+export function doubleclickIsA4AEnabled(win, element, useRemoteHtml) {
+  return singleton.isA4aEnabled(win, element, useRemoteHtml);
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: you could be fancy and just apply the arguments here to ensure that as parameters are added you don't need to do anything... (though you loose type check).

File: extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js [hunk 7]
```diff
@@ -19,10 +19,14 @@ const SRC_PREFIX_ = 'https://ib.3lift.com/';
 /**
  * @param {!Window} win
  * @param {!Element} element
+ * @param {!boolean} useRemoteHtml
  * @returns {boolean}
  */
-export function tripleliftIsA4AEnabled(win, element) {
+export function tripleliftIsA4AEnabled(win, element, useRemoteHtml) {
+  if (useRemoteHtml) {
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js]: Remove

File: extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js [hunk 8]
```diff
@@ -19,10 +19,14 @@ const SRC_PREFIX_ = 'https://ib.3lift.com/';
...
+    return false;
+  }
   let src;
-  return !!element.getAttribute('data-use-a4a') && !!(src =
+  return !useRemoteHtml && !!element.getAttribute('data-use-a4a') && !!(src =
     element.getAttribute('src')) && src.indexOf(SRC_PREFIX_) == 0;
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js]: Could use src/string#startsWith

File: extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js (hunk 7 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-triplelift-impl/0.1/triplelift-a4a-config.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js (hunk 6 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: not sure if I understand what you mean

**Pull request merged**